design/ceph/object/read-affinity.md

```diff
@@ -0,0 +1,92 @@
+# Ceph RGW Read Affinity
+
+
+## Overview
+- RGW clients can read data from any replica without any prioritization. This may have cost implications due to cross-zone traffic distribution.
+- In order to optimize network performance and reduce data transfer costs:
+    - Client read requests should preferably be sent to the RGW daemon that is in the same zone as client. Refer Frontend Data Locality section below.
+    - RGW daemon should send the read requests to the nearest possible OSD. Refer Backend Data Locality section below.
+
+
+## Frontend Data Locality
+- Users communicate with the Ceph Object Storage via K8s service that abstracts RGW daemons PODs.
+- By default, the traffic is distributed evenly across all the available RGW daemon POD endpoints, regardless of their location relative to the users.
+- `TrafficDistribution` can attempt to route traffic to the closest healthy endpoints.
+
+### TrafficDistribution
+- With `spec.trafficDistribution.PreferClose`, the endpoints within a zone will receive all the traffic for that zone. If there are no endpoints in a zone, traffic will be distributed to other zones.
+
+
+### Implementation:
+
+#### `API Changes`
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStore
+metadata:
+  name: my-store
+  namespace: rook-ceph
+spec:
+  gateway:
+    readAffinnity:
+        type: localize
+        failureDomain: zone
+        enableTrafficDistribution: true
```

Comment:
Alternatively, I think it might be helpful for users to create a config section where they can pass in service options -- we might need/want this in the future as well. I think that'll make it easy to implement and document since it'll be a simple passthru.

```yaml
service: # serviceSpec, serviceOptions, other naming?
  trafficDistribution: PreferClose
```